File: ABC/abc212/src/bin/c.rs
use proconio::input;

fn main() {
    input! {
        n: usize,
        m: usize,
        mut a: [i32; n],
        mut b: [i32; m],
    }

    a.sort();
    b.sort();

    let mut ai = 0;
    let mut bi = 0;
    let mut res = i32::MAX;
    while ai < n && bi < m {
        let diff = (a[ai] - b[bi]).abs();
        res = res.min(diff);
        if a[ai] < b[bi] { ai += 1 } else { bi += 1 }
    }

    println!("{}", res);
}

File: ABC/abc212/c.py
from bisect import bisect

n, m = map(int, input().split())
a = list(map(int, input().split()))
b = list(map(int, input().split()))
a.sort()
b.sort()

ans = 10**9
for num in a:
    idx = bisect(b, num)

    if idx < m:
        ans = min(ans, abs(num - b[idx]))
    if idx > 0:
        ans = min(ans, abs(num - b[idx - 1]))

print(ans)
